File: services/trade_analytics.py
from sqlalchemy.orm import Session
from database.models import TradePerformance

def get_trade_performance(user_id: int, db: Session):
    """Fetch trade history and profit/loss for the analytics dashboard."""
    
    trades = db.query(TradePerformance).filter(TradePerformance.user_id == user_id).all()
    trade_data = []
    
    for trade in trades:
        trade_data.append({
            "symbol": trade.symbol,
            "trade_type": trade.trade_type,
            "entry_price": trade.entry_price,
            "exit_price": trade.exit_price if trade.exit_price else "Open Trade",
            "profit_loss": trade.profit_loss if trade.profit_loss else "Pending",
            "trade_time": trade.trade_time,
            "status": trade.status
        })
    
    return trade_data

File: ui/trading-bot-ui/src/components/TradeAnalytics.js
import { useEffect, useState } from "react";
const BASE_URL = process.env.REACT_APP_API_URL;
function TradeAnalytics({ userId }) {
    const [trades, setTrades] = useState([]);

    useEffect(() => {
        fetch(`${BASE_URL}/api/analytics?user_id=${userId}`)
            .then(response => response.json())
            .then(data => setTrades(data))
            .catch(error => console.error("Error fetching trade analytics:", error));
    }, [userId]);

    return (
        <div>
            <h2>Trade Analytics</h2>
            <table>
                <thead>
                    <tr>
                        <th>Symbol</th>
                        <th>Trade Type</th>
                        <th>Entry Price</th>
                        <th>Exit Price</th>
                        <th>Profit/Loss</th>
                        <th>Status</th>
                    </tr>
                </thead>
                <tbody>
                    {trades.map((trade, index) => (
                        <tr key={index}>
                            <td>{trade.symbol}</td>
                            <td>{trade.trade_type}</td>
                            <td>₹{trade.entry_price}</td>
                            <td>{trade.exit_price === "Open Trade" ? "Open" : `₹${trade.exit_price}`}</td>
                            <td style={{ color: trade.profit_loss >= 0 ? "green" : "red" }}>
                                {trade.profit_loss === "Pending" ? "Pending" : `₹${trade.profit_loss}`}
                            </td>
                            <td>{trade.status}</td>
                        </tr>
                    ))}
                </tbody>
            </table>
        </div>
    );
}

export default TradeAnalytics;
